### DISCUSSION TITLE
Docker file for custom binaries

### DISCUSSION THREAD

**Discussion by tigawanna** [pocketbase/pocketbase#1650] [Q&A] [answered]
Hi am kind of new to GO and docker and can't quite get my image to build without issues 

```Dockerfile
FROM alpine:latest

# ARG PB_VERSION=0.10.1

RUN apk add --no-cache \
    # unzip \
    # this is needed only if you want to use scp to copy later your pb_data locally
    openssh

# download and unzip PocketBase
ADD https://github.com/tigawanna/devhub-backend/blob/master/pocketbase /pb/pocketbase
# RUN unzip /tmp/pb.zip -d /pb/
RUN chmod 777 /pb/pocketbase
EXPOSE 8080


# start PocketBase
CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8080"]
```
am getting a wrong exec format error and insufficient permissions errors hence the chmod 777, I prebuilt it for linux_amd_64 before pushing it. 
is there a better way to do this?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand what you are trying to do.

It seems that the above downloads a prebuilt executable from your github repo and tries to run it.
If that's what you want, you don't usually need to make it 777, just the `+x` flag should be enough.

Also the url you are using doesn't seem to point to the file, but the GitHub file explorer. You have to provide a direct download link, something like:
```
https://github.com/tigawanna/devhub-backend/blob/master/pocketbase?raw=true
```

If that's not the issue, make sure that you have `CGO_ENABLED=0` so that you can create a statically linked binary:

```sh
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build
```

> is there a better way to do this?

You can build the binary inside your docker image (_you can also use a multi-stage image_).
For a simple example you can check the golang image readme - https://hub.docker.com/_/golang.
